File: opinionpulse/opinionpulse-frontend/src/api/quiver.ts
import { apiRequest } from "@/api/client"

export type QuiverRecord = {
  title: string
  subtitle: string | null
  date: string | null
  detail: string | null
  amount: string | null
  meta: Record<string, string>
}

export type QuiverSection = {
  id: string
  label: string
  emoji: string
  description: string
  available: boolean
  message: string | null
  records: QuiverRecord[]
}

export type QuiverIntelligenceResponse = {
  query: string
  ticker: string | null
  configured: boolean
  asset_type: "stock" | "crypto" | "unknown"
  sections: QuiverSection[]
  source: string
  message: string | null
}

export async function getQuiverIntelligence(query: string): Promise<QuiverIntelligenceResponse> {
  const params = new URLSearchParams({ q: query })
  return apiRequest<QuiverIntelligenceResponse>(`/api/market/quiver?${params.toString()}`, {
    auth: true,
  })
}

File: opinionpulse/opinionpulse-backend/app/schemas/quiver.py
from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel, Field


class QuiverRecord(BaseModel):
    title: str
    subtitle: Optional[str] = None
    date: Optional[str] = None
    detail: Optional[str] = None
    amount: Optional[str] = None
    meta: dict[str, str] = Field(default_factory=dict)


class QuiverSection(BaseModel):
    id: str
    label: str
    emoji: str
    description: str
    available: bool = True
    message: Optional[str] = None
    records: list[QuiverRecord] = Field(default_factory=list)


class QuiverIntelligenceResponse(BaseModel):
    query: str
    ticker: Optional[str] = None
    configured: bool
    asset_type: Literal["stock", "crypto", "unknown"]
    sections: list[QuiverSection] = Field(default_factory=list)
    source: str = "Quiver Quantitative"
    message: Optional[str] = None
